### PR TITLE
JetPerformance: Add option to plot a couple trigger response histos in MC

### PR DESCRIPTION
**Pull request by jdmulligan**
(update to my user task)

@mfasDa I would appreciate if you take a look

### CODE REVIEW

File: PWGJE/EMCALJetTasks/UserTasks/AliAnalysisTaskEmcalJetPerformance.cxx [hunk 1]
```diff
@@ -1301,6 +1322,70 @@ void AliAnalysisTaskEmcalJetPerformance::DoTriggerSimulation()
...
+    AliTLorentzVector part;
+    Double_t partEta;
+    Double_t partPhi;
+    Double_t partE;
+    for (auto mcpart:fGeneratorLevel->accepted_momentum()) {
+      part.Clear();
+      part = mcpart.first;
+      partEta = part.Eta();
+      partPhi = part.Phi_0_2pi();
+      partE = part.E();
```
Comment (mfasDa) [inline PWGJE/EMCALJetTasks/UserTasks/AliAnalysisTaskEmcalJetPerformance.cxx]: From what I understand you sum up the energy of all particles including hadrons. Hadrons however don't deposit the full energy in the EMCAL. You either have to apply an effective hadronic correction similar to the correction applied in the hadronic correction component or you take the parameterized response from Mateusz. Only for electrons and photons you should assume full energy deposit in the EMCAL. Note this is only relevant at generator level, at cluster level the detector response has already been applied.

**Comment by mfasDa** [comment]
Hi @jdmulligan ,

I have just one comment which is that at generator level the energy has to be corrected for everything else than photons and electrons, otherwise the patch energy will be significantly overestimated as normally it only contains the neutral energy part of the jet. Hadrons only deposit part of their energy in the detector. There are effective detector responses available for pions (muons), kaons and protons, you should apply them.

Cheers

Markus

**Comment by jdmulligan** [comment]
Hi @mfasDa,

My understanding was that these effects would be taken care of on Mateusz's side, and from me he just wants the response to the total truth energy. I'll check.

Cheers,
James

**Comment by mfasDa** [comment]
+1

**Pull request merged**